benchexec/containerexecutor.py

```diff
@@ -713,11 +713,28 @@ def child():
                         try:
                             socket.sethostname(container.CONTAINER_HOSTNAME)
                         except PermissionError:
-                            logging.warning(
-                                "Changing hostname in container prevented "
-                                "by system configuration, "
-                                "real hostname will leak into the container."
-                            )
+                            # Ubuntu 24.04 (and possibly later versions) restricts user namespaces,
+                            # output error message here
+                            if (
+                                util.try_read_file(
+                                    "/proc/sys/kernel/apparmor_restrict_unprivileged_userns"
+                                )
+                                == "1"
+                            ):
+                                logging.warning(
+                                    "Ubuntu 24.04 restircts user namespaces, "
+                                    "preventing changing hostname in container, "
+                                    "real hostname will leak into the container. "
+                                    "Please try 'echo 0 | sudo tee "
+                                    "/proc/sys/kernel/apparmor_restrict_unprivileged_userns' "
+                                    "as a temporary workaround, or disable container mode."
```

Comment:
For such long and deeply indented log messages it can be useful to define them in a module-level constant, where the usable parts of lines are longer and the message is thus easier to read.